front/app/components/EsriMap/config.ts

```diff
@@ -21,12 +23,20 @@ export const configureMapView = (
   setMapCenter(mapView, initialData, globalMapSettings);
 
   // Set initial extent
-  mapView.zoom = initialData?.zoom || globalMapSettings.zoom_level || 18;
+  const zoomLevel = initialData?.zoom || globalMapSettings.zoom_level || 18;
+  mapView.zoom = zoomLevel;
   mapView.constraints = {
     maxZoom: initialData?.maxZoom || 22,
     minZoom: 5,
   };
 
+  // If the map is not in the standard Web Mercator projection (3857) or WGS 84 (4326),
+  // ensure that the map scale is set based on the zoom, so the zoom level works as intended.
+  const spatialReferenceId = mapView.center.spatialReference.wkid;
+  if (spatialReferenceId !== 3857 && spatialReferenceId !== 4326) {
+    mapView.scale = calculateScaleFromZoom(zoomLevel);
+  }
+
```

Comment:
Vast majority of our clients use these well known map projections (3857 or 4326) for their maps which work great.

For any client using something else though (like Lahti), we should also just explicitly set this MapView.Scale just in case it's one of the map types which only uses the "Map Scale" like I mentioned above.